cfme/tests/infrastructure/test_vmware_provider.py

```diff
@@ -98,7 +99,10 @@ def test_vmware_vds_ui_display(appliance, provider):
     except IndexError:
         pytest.skip("No hosts found")
     view = navigate_to(host, 'Networks')
-    assert 'DSwitch' in view.network_tree.all_options
+    soft_assert('DSwitch' in view.network_tree.all_options, "No DSwitches on Host Network page")
+    switches_collection = appliance.collections.infra_switches
+    switches = [switch.name for switch in switches_collection.all() if switch.name == 'DSwitch']
```

Comment:
I think this would be a bit cleaner as the following - note no soft_assert because its the last assertion in the test, and isn't in a loop.

`assert 'DSwitch' in [s.name for s in appliance.collections.infra_switches.all()], 'No DSwitches on networking page`